crates/turborepo-api-client/src/lib.rs

```diff
@@ -376,10 +377,10 @@ impl Client for APIClient {
 
         let response = retry::make_retryable_request(request_builder).await?;
 
-        if response.status() == StatusCode::FORBIDDEN {
-            Err(Self::handle_403(response).await)
-        } else {
-            Ok(response.error_for_status()?)
+        match response.status() {
+            StatusCode::FORBIDDEN => Err(Self::handle_403(response).await),
```

Comment:
Had to tweak this because we weren't actually producing cache miss errors. In the future I'd like to refactor cache misses to not be an error